substrate/frame/vesting/src/mock.rs

```diff
@@ -151,7 +151,10 @@ impl ExtBuilder {
 			.assimilate_storage(&mut t)
 			.unwrap();
 		let mut ext = sp_io::TestExternalities::new(t);
-		ext.execute_with(|| System::set_block_number(1));
+		ext.execute_with(|| {
+			System::set_block_number(1);
+			Vesting::do_try_state().unwrap();
+		});
```

Comment:
This seems to be running the try_state checks when building the externalities. I think it would be more useful to run the try state checks at the *end* of the test, to ensure that all the invariants hold true after the different test cases.

We can refactor a bit the `ExtBuilder` implementation to do this and implement a `fn build_and_execute` method that builds the ext, runs the test and then calls the try_state. Something along these lines:

```rust
pub fn build_and_execute(self, test: impl FnOnce() -> ()) {
		sp_tracing::try_init_simple();
		let mut ext = self.build();
		ext.execute_with(test);
		ext.execute_with(|| {
			Vesting::do_try_state().unwrap();
		});
	}
``` 

We then can change the tests to execute the tests under `ExtBuilder::default().build_and_execute(|| {..})` instead of explicitly calling `build().execute()` as it is now. That way we don't need to sprinkle the try_state calls all around the tests. wdyt?

Reply:
It is a much better approach, already made the changes